From 400e86ef933b80e70f8eee237e613cb7121fbe66 Mon Sep 17 00:00:00 2001
From: dponna534 <Deepak_Ponnath2@comcast.com>
Date: Mon, 18 Aug 2025 15:43:33 +0530
Subject: [PATCH] RDKEMW-3659: Fix WPE_ERROR_ResourceUnavail message on
 getModel

Reason for change: Fix WPE_ERROR_ResourceUnavail message on getModel
Test Procedure: Refer ticket
Risks: Low

Signed-off-by: Deepak_Ponnath2@comcast.com
---
 SystemServices/SystemServices.cpp |  40 +++++++++-
 SystemServices/SystemServices.h   |   1 +
 helpers/UtilsController.h         | 124 ------------------------------
 3 files changed, 40 insertions(+), 125 deletions(-)

diff --git a/SystemServices/SystemServices.cpp b/SystemServices/SystemServices.cpp
index 703db5f..a8b21a8 100755
--- a/SystemServices/SystemServices.cpp
+++ b/SystemServices/SystemServices.cpp
@@ -69,6 +69,13 @@
 #include "UtilsfileExists.h"
 #include "UtilsgetFileContent.h"
 #include "UtilsProcess.h"
+#include "UtilsController.h"
+
+#ifdef USE_THUNDER_R4
+#include <interfaces/IDeviceInfo.h>
+#else
+#include <interfaces/IDeviceInfo2.h>
+#endif /* USE_THUNDER_R4 */
 
 using namespace std;
 using namespace WPEFramework;
@@ -2438,7 +2445,7 @@ namespace WPEFramework {
             std::string estbMac = collectDeviceInfo("estb_mac");
             removeCharsFromString(estbMac, "\n\r");
             rConf["eStbMac"] = estbMac;
-            rConf["model"] = getModel();
+            rConf["model"] = RetrieveModelNumberThroughCOMRPC();
             rConf["firmwareVersion"] = stbVersion;
             response["xconfParams"] = rConf;
             returnResponse(true);
@@ -4417,6 +4424,37 @@ namespace WPEFramework {
             return "unknown";
 #endif
         }
+
+	string SystemServices::RetrieveModelNumberThroughCOMRPC()
+	{
+	    LOGINFO("RetrieveModelNumberThroughCOMRPC Entry\n");
+	    std::string Number;
+	    if (m_shellService)
+	    {
+	       PluginHost::IShell::state state;
+
+	       if ((Utils::getServiceState(m_shellService, "DeviceInfo", state) == Core::ERROR_NONE) && (state != PluginHost::IShell::state::ACTIVATED))
+	       {
+		  Utils::activatePlugin(m_shellService, "DeviceInfo");
+	       }
+	       if ((Utils::getServiceState(m_shellService, "DeviceInfo", state) == Core::ERROR_NONE) && (state == PluginHost::IShell::state::ACTIVATED))
+	       {
+		  auto _remoteDeviceInfoObject = m_shellService->QueryInterfaceByCallsign<Exchange::IDeviceInfo>("DeviceInfo");
+
+		  if(_remoteDeviceInfoObject)
+		  {
+		    _remoteDeviceInfoObject->Sku(Number);
+		    _remoteDeviceInfoObject->Release();
+		  }
+		}
+		else
+		{
+		    LOGERR("Failed to create DeviceInfo object\n");
+		}
+	    }
+	     return Number;
+	}
+
 	string SystemServices::getStbBranchString()
 	{
 		static string stbBranchStr;
diff --git a/SystemServices/SystemServices.h b/SystemServices/SystemServices.h
index 0c09994..3f50306 100644
--- a/SystemServices/SystemServices.h
+++ b/SystemServices/SystemServices.h
@@ -258,6 +258,7 @@ namespace WPEFramework {
                 std::string getStbTimestampString();
 		std::string getStbBranchString();
                 bool makePersistentDir();
+                std::string RetrieveModelNumberThroughCOMRPC();
 
 #if defined(USE_IARMBUS) || defined(USE_IARM_BUS)
                 void InitializeIARM();
diff --git a/helpers/UtilsController.h b/helpers/UtilsController.h
index f3729bb..1975e18 100644
--- a/helpers/UtilsController.h
+++ b/helpers/UtilsController.h
@@ -22,141 +22,17 @@
 #include <mutex>
 #include <curl/curl.h>
 
-#ifndef DISABLE_SECURITY_TOKEN
-#include <securityagent/SecurityTokenUtil.h>
-#endif
 
 // std
 #include <string>
 
-#define MAX_STRING_LENGTH 2048
 
-#define SERVER_DETAILS  "127.0.0.1:9998"
 
 using namespace WPEFramework;
 using namespace std;
 
 namespace Utils
 {
-    struct SecurityToken
-    {
-        static void getSecurityToken(std::string &token)
-        {
-            static std::string sToken = "";
-            static bool sThunderSecurityChecked = false;
-
-            static std::mutex mtx;
-            std::unique_lock<std::mutex> lock(mtx);
-
-            if (sThunderSecurityChecked)
-            {
-                token = sToken;
-                return;
-            }
-
-            sThunderSecurityChecked = true;
-
-#ifdef DISABLE_SECURITY_TOKEN
-            token = sToken;
-#else
-            if (!isThunderSecurityConfigured())
-            {
-                LOGINFO("Thunder Security is not enabled. Not getting token");
-                token = sToken;
-                return;
-            }
-
-            unsigned char buffer[MAX_STRING_LENGTH] = {0};
-            int ret = GetSecurityToken(MAX_STRING_LENGTH, buffer);
-            if (ret < 0)
-            {
-                LOGERR("Error in getting token");
-            }
-            else
-            {
-                LOGINFO("Retrieved token successfully");
-                token = (char *)buffer;
-                sToken = token;
-            }
-#endif
-        }
-
-#ifndef DISABLE_SECURITY_TOKEN
-        static size_t writeCurlResponse(void *ptr, size_t size, size_t nmemb, string stream)
-        {
-            size_t realsize = size * nmemb;
-            string temp(static_cast<const char *>(ptr), realsize);
-            stream.append(temp);
-            return realsize;
-        }
-
-        static bool isThunderSecurityConfigured()
-        {
-            bool configured = false;
-            long http_code = 0;
-            std::string jsonResp;
-            CURL *curl_handle = NULL;
-            CURLcode res = CURLE_OK;
-            curl_handle = curl_easy_init();
-            string serialNumber = "";
-            string url = "http://127.0.0.1:9998/Service/Controller/Configuration/Controller";
-            if (curl_handle &&
-                !curl_easy_setopt(curl_handle, CURLOPT_URL, url.c_str()) &&
-                !curl_easy_setopt(curl_handle, CURLOPT_HTTPGET, 1) &&
-                !curl_easy_setopt(curl_handle, CURLOPT_FOLLOWLOCATION, 1) && // when redirected, follow the redirections
-                !curl_easy_setopt(curl_handle, CURLOPT_WRITEFUNCTION, writeCurlResponse) &&
-                !curl_easy_setopt(curl_handle, CURLOPT_WRITEDATA, &jsonResp))
-            {
-
-                res = curl_easy_perform(curl_handle);
-                if (curl_easy_getinfo(curl_handle, CURLINFO_RESPONSE_CODE, &http_code) != CURLE_OK)
-                {
-                    std::cout << "curl_easy_getinfo failed\n";
-                }
-                std::cout << "Thunder Controller Configuration ret: " << res << " http response code: " << http_code << std::endl;
-                curl_easy_cleanup(curl_handle);
-            }
-            else
-            {
-                std::cout << "Could not perform curl to read Thunder Controller Configuration\n";
-            }
-            if ((res == CURLE_OK) && (http_code == 200))
-            {
-                // check for "Security" in response
-                JsonObject responseJson = JsonObject(jsonResp);
-                if (responseJson.HasLabel("subsystems"))
-                {
-                    const JsonArray subsystemList = responseJson["subsystems"].Array();
-                    for (int i = 0; i < subsystemList.Length(); i++)
-                    {
-                        string subsystem = subsystemList[i].String();
-                        if (subsystem == "Security")
-                        {
-                            configured = true;
-                            break;
-                        }
-                    }
-                }
-            }
-            return configured;
-        }
-#endif
-  
-    };
-
-    // Thunder Plugin Communication
-    std::shared_ptr<WPEFramework::JSONRPC::LinkType<WPEFramework::Core::JSON::IElement>> getThunderControllerClient(std::string callsign="")
-    {
-
-        string token;
-        Utils::SecurityToken::getSecurityToken(token);
-        string query = "token=" + token;
-
-        Core::SystemInfo::SetEnvironment(_T("THUNDER_ACCESS"), (_T(SERVER_DETAILS)));
-        std::shared_ptr<WPEFramework::JSONRPC::LinkType<WPEFramework::Core::JSON::IElement>> thunderClient = make_shared<WPEFramework::JSONRPC::LinkType<WPEFramework::Core::JSON::IElement>>(callsign.c_str(), "", false, query);
-
-        return thunderClient;
-    }
 
 #ifndef USE_THUNDER_R4
     class Job : public Core::IDispatchType<void>
-- 
2.47.1.windows.2

